lib/dfe/analytics/railtie.rb

```diff
@@ -34,7 +34,15 @@ class Railtie < Rails::Railtie
       config.after_initialize do
         # internal gem tests will sometimes suppress this so they can test the
         # init process
-        DfE::Analytics.initialize! unless ENV['SUPPRESS_DFE_ANALYTICS_INIT']
+        if running_db_rake_task? || ENV['SUPPRESS_DFE_ANALYTICS_INIT']
+          puts 'Skipping DfE::Analytics initialization'
+        else
+          DfE::Analytics.initialize!
+        end
+      end
+
+      def running_db_rake_task?
+        defined?(Rake) && Rake.application.top_level_tasks.any? { |t| t.start_with?('db:') }
```

Comment:
I was trying to see if there was a way we could look at the namespace of the tasks rather than looking at what it starts with, but couldn't work out a nice way of doing it, so this seems fine.

Reply:
Yeah, it's a bit ick. But it does seem to work